cpp/src/arrow/compute/exec/exec_plan.h

```diff
@@ -146,6 +148,46 @@ class ARROW_EXPORT ExecNode {
 
   virtual Status Validate() const;
 
+  /// \brief the ordering of the output batches
+  ///
+  /// This does not guarantee the batches will be emitted by this node
+  /// in order.  Instead it guarantees that the batches will have their
+  /// ExecBatch::index property set in a way that respects this ordering.
+  ///
+  /// In other words, given the ordering {{"x", SortOrder::Ascending}} we
+  /// know that all values of x in a batch with index N will be less than
+  /// or equal to all values of x in a batch with index N+k (assuming k > 0)
```

Comment:
If there is an order based on a column "x", does that also guarantee something about the order _within_ each batch? (or only between batches as this paragraph explains)

Reply:
Yes, it should guarantee that the ordering exists within the batch as well.

Reply:
I've updated the wording here to mention ordering within a batch.